client/payments/data/actions.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { apiFetch } from '@wordpress/data-controls';
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES } from './action-types';
+import { API_NAMESPACE } from './constants';
+import { PaymentGateway, RestApiError, SelectorKeysWithActions } from './types';
+
+export function getPaymentGatewaysRequest(): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_REQUEST;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_REQUEST,
+	};
+}
+
+export function getPaymentGatewaysSuccess(
+	paymentGateways: PaymentGateway[]
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_SUCCESS;
+	paymentGateways: PaymentGateway[];
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_SUCCESS,
+		paymentGateways,
+	};
+}
+
+export function getPaymentGatewaysError(
+	error: RestApiError
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_ERROR;
+	error: RestApiError;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_ERROR,
+		error,
+	};
+}
+
+export function getPaymentGatewayRequest(): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_REQUEST;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_REQUEST,
+	};
+}
+
+export function getPaymentGatewayError(
+	error: RestApiError
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_ERROR;
+	error: RestApiError;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_ERROR,
+		error,
+	};
+}
+
+export function updatePaymentGatewaySuccess(
+	paymentGateway: PaymentGateway,
+	selector: 'updatePaymentGateway' | 'getPaymentGateway'
```

Comment:
I don't think we need this `selector` here anymore since we can derive from the event `UPDATE_PAYMENT_GATEWAY_SUCCESS` that it should be `updatePaymentGateway`.